src/main/kotlin/com/alecstrong/grammar/kit/composer/GrammarKitComposerPlugin.kt

```diff
@@ -2,23 +2,26 @@ package com.alecstrong.grammar.kit.composer
 
 import org.gradle.api.Plugin
 import org.gradle.api.Project
+import org.gradle.api.file.Directory
 import org.gradle.api.tasks.SourceSetContainer
 import org.jetbrains.grammarkit.GrammarKitPlugin
+import org.jetbrains.grammarkit.path
 import org.jetbrains.grammarkit.tasks.GenerateParserTask
 import java.io.File
 
 open class GrammarKitComposerPlugin : Plugin<Project> {
   override fun apply(project: Project) {
     project.pluginManager.apply(GrammarKitPlugin::class.java)
-    project.file("src${File.separatorChar}main${File.separatorChar}kotlin").forBnfFiles { bnfFile ->
-      val rootDir = project.file("src${File.separatorChar}main${File.separatorChar}kotlin")
-      val name = bnfFile.toRelativeString(rootDir).replace(File.separatorChar, '_').dropLast(4)
-      val outputDirectory = File(project.buildDir, "grammars${File.separatorChar}$name")
+
+    val rootDir = project.layout.projectDirectory.dir("src${File.separatorChar}main${File.separatorChar}kotlin")
+    rootDir.forBnfFiles { bnfFile ->
+      val name = bnfFile.toRelativeString(rootDir.asFile).replace(File.separatorChar, '_').dropLast(4)
+      val outputDirectory = project.layout.buildDirectory.dir("grammars${File.separatorChar}$name")
 
       val compose = project.tasks.register("createComposable${name}Grammar", BnfExtenderTask::class.java) {
-        it.source(bnfFile)
-        it.outputDirectory = outputDirectory
-        it.include("**${File.separatorChar}*.bnf")
```

Comment:
Do you know the reason for `include('**/*.bnf')` in the bnfExtenderTask?

Reply:
incremental compilation, including the files as source is how it knows when to re-run the task

Reply:
But we create 1 task for each bnf file by setting the `source` property. Why do we need to set include too? The input is only 1 bnf file. As long as we don't use multiple bnf files per task removing this code is fine.